app/src/main/java/org/gdg/frisbee/android/view/LocationListPreference.java

```diff
@@ -0,0 +1,290 @@
+package org.gdg.frisbee.android.view;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.preference.DialogPreference;
+import android.preference.PreferenceManager;
+import android.text.TextUtils;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.Adapter;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Filter;
+import android.widget.ListView;
+import android.widget.SearchView;
+
+import org.gdg.frisbee.android.R;
+
+public class LocationListPreference extends DialogPreference implements AdapterView.OnItemClickListener {
+
+    private FilterListView listView;
+    private int clickedItemIndex;
+    private String mValue;
+    private boolean mValueSet;
+    private String[] mEntries;
+    private String[] mEntryValues;
+
+    public LocationListPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setDialogLayoutResource(R.layout.view_location_list_preference);
+    }
+
+    @Override
+    protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
+        super.onPrepareDialogBuilder(builder);
+        builder.setPositiveButton(null, null);
+    }
+
+    @Override
+    protected void onBindDialogView(View view) {
+        super.onBindDialogView(view);
+        listView = (FilterListView) view.findViewById(android.R.id.list);
```

Comment:
I think there is too much things going on here. Is it possible to extend directly from `ListPreference`. I am not sure but if it can be done that way, the list inside ListPreference can be used. It supports single choice mode and everything by default.